### PR TITLE
Add one more generic type for TypeScript users

### PR DESCRIPTION
**Pull request by dusansimic**
Adding another generic type to the functions allows developer to easily
specify the data type he wants to receive and return to the emitter.

Closes: https://github.com/sindresorhus/electron-better-ipc/issues/23

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -25,11 +25,11 @@ export interface MainProcessIpc extends IpcMain {
 	})();
 	```
 	*/
-	callRenderer<T>(
+	callRenderer<T, S>(
```
Comment (sindresorhus) [inline index.d.ts]: I think we should use more descriptive type names than `T` and `S` now that there are multiple.

**Comment by sindresorhus** [comment]
Can you also add support for https://github.com/sindresorhus/electron-better-ipc/commit/3112e7be83eb03e6359beb83fce10f2fc0be8286 ?

**Comment by sindresorhus** [comment]
Would be good to have a few type tests for this: https://github.com/sindresorhus/electron-better-ipc/blob/master/index.test-d.ts

**Comment by dusansimic** [comment]
Added all requested changes. Does everything look ok?

**Comment by dusansimic** [comment]
I found a bug while working on the issue mentioned in the description of PR and now fixed it with this small change. The problem was in the type of arguments passed to `callRenderer` and type of parameters of `answerMain`. The thing is TypeScript automatically adds a `undefined` type to a parameter if its specified as optional so that made a problem with `callRenderer` and `answerMain` functions as there was an error that argument and parameter types were not compatible as shown in the image.
image
This problem is now fixed and everything is ready for review.

**Comment by UberMouse** [comment]
Any news on this? I've switched to pulling down directly off @dusansimic fork so I can get access to the typings. Would love to be able to switch back to the normal NPM package

**Pull request merged**